css/css-ui-3/text-overflow-013.html

```diff
@@ -0,0 +1,41 @@
+﻿<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Basic User Interface Test: ellipsis and baseline alignement</title>
+<link rel="author" title="Florian Rivoal" href="http://florian.rivoal.net/">
+<link rel="help" href="http://www.w3.org/TR/css3-ui/#ellipsing-details">
+<link rel="match" href="../reference/ref-filled-green-100px-square.xht">
+<meta name="flags" content="ahem">
+<meta name="assert" content="The ellipsis is be baseline aligned with the block, even if the thing it replaces isn't.">
+<style>
+.test {
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: pre;
+  line-height: 1;
+  font-size: 100px;
+  font-family: ahem;
+  color: green;
+
+  /* 2em width and starting with a space
+     because the handling of the ellipsis of the first character on the line is special,
+     and therefore we need to have the thing we want to test in non first position.
+     Shifting 1em left to align with the usual green square.
+   */
+  width: 2em;
+  margin-left: -1em;
+}
+.red {
+  position: absolute;
+  z-index: -1;
+  width: 100px;
+  height: 100px;
+  background: red;
+}
+span {
+  vertical-align: sub;
+}
+</style>
+
+<p>Test passes if there is a filled green square and <strong>no red</strong>.</p>
+<div class=red></div>
+<div class=test> <span> xx</span> </div>
```

Comment:
Overall not a bad test, but ...
  - Let's make elided text red and visible so that if it isn't elided the test fails.
  - Ensure that the elided text and the characters before and after it are all encased in the span so that the ellipsis can't be referencing any adjacent direct contents of the block.
  - Also put a line break within the span, both before and after the line being tested, so that the entire line box being tested unquestionably contains only the contents of the span. This way it's clear that it is referencing the baseline of the block even when the block has no direct content on the line. (You can also have elided text on the lines before/after this particular test if you want to test combination lines, but we should have at least one test where there is no direct block content on the line.)